vms/platformvm/state/chain_time_helpers.go

```diff
@@ -77,5 +77,16 @@ func GetNextStakerChangeTime(state Chain) (time.Time, error) {
 // PickFeeCalculator does not modify [state].
 func PickFeeCalculator(cfg *config.Config, state Chain) (fee.Calculator, error) {
 	timestamp := state.GetTimestamp()
-	return fee.NewStaticCalculator(cfg.StaticFeeConfig, cfg.UpgradeConfig, timestamp), nil
+	return NewStaticFeeCalculator(cfg, timestamp), nil
+}
+
+// NewStaticFeeCalculator creates a static fee calculator, with the config set
+// to either the pre-AP3 or post-AP3 config.
+func NewStaticFeeCalculator(cfg *config.Config, timestamp time.Time) fee.Calculator {
+	config := cfg.StaticFeeConfig
+	if !cfg.UpgradeConfig.IsApricotPhase3Activated(timestamp) {
+		config.CreateSubnetTxFee = cfg.CreateAssetTxFee
+		config.CreateBlockchainTxFee = cfg.CreateAssetTxFee
+	}
+	return fee.NewStaticCalculator(config)
 }
```

Comment:
I'm not convinced that this is the right place for these methods... But will leave that decision to a later PR.